src/backend/distributed/citus--7.1-2--7.1-3.sql

```diff
@@ -0,0 +1,10 @@
+/* citus--7.1-2--7.1-3 */
+
+CREATE TABLE citus.pg_dist_metadata(
+    tag text NOT NULL,
+    value text NOT NULL
+);
+ALTER TABLE citus.pg_dist_metadata SET SCHEMA pg_catalog;
+GRANT SELECT ON pg_catalog.pg_dist_metadata TO public;
+
+INSERT INTO pg_dist_metadata VALUES ('server_id', (random() * 1e18)::bigint::text);
```

Comment:
Thanks for the research on the seeding process. I'm comfortable with everything you found.

I guess I wasn't entirely clear about using `random` as it is used in the `uint` function… I didn't mean using it in SQL-land, since getting at the byte representation of e.g. a `float` in that environment is tough. I kind of meant just writing our own C function to precisely do what we want and using it.

After thinking about that some more, I realized SQL doesn't even _have_ a `uint` type. I looked at the code to generate type 4 (i.e. pure-random) UUIDs and found it was very simple. Can we just do something like this (a modified `pg_random_uuid` from `pgcrypto`)?

```c
PG_FUNCTION_INFO_V1(citus_server_id);

Datum
citus_server_id(PG_FUNCTION_ARGS)
{
	const Size uuidLength = 16;
	uint8 *randomBytes = (uint8 *) palloc(uuidLength);

	if (!pg_backend_random((char *) randomBytes, uuidLength))
	{
		ereport(ERROR, (errmsg("failed to generate server identifier")))
	}

	/*
	 * Set magic numbers for a "version 4" (pseudorandom) UUID, see
	 * http://tools.ietf.org/html/rfc4122#section-4.4
	 */
	buf[6] = (buf[6] & 0x0f) | 0x40;	/* "version" field */
	buf[8] = (buf[8] & 0x3f) | 0x80;	/* "variant" field */

	PG_RETURN_UUID_P((pg_uuid_t *) buf);
}
```

Then we know what we're getting is a good, random UUID and we can just cast it to text when inserting it to the `jsonb` value.